gluster-exporter/metric_brick.go

```diff
@@ -131,14 +132,14 @@ func diskUsage(path string) (disk DiskStatus, err error) {
 func brickUtilization() {
```

Comment:
one idea to avoid multiple logging here:

change this function to `func brickUtilization() error`. On failure it returns error to caller(`main.go`), handle error in `main.go` and log error.

```
go func(m glusterMetric) {
	for {
		err = m.fn()
		interval := defaultInterval
		if collectorConf.SyncInterval > 0 {
			interval = time.Duration(collectorConf.SyncInterval)
		}
		if err != nil {
			log.WithError(err).WithFields(log.Fields{
				"name": m.name,
			}).Error("failed to export metric")
		}
		time.Sleep(time.Duration(time.Second * time.Duration(interval)))
	}
}(m)
```

Advantage with this approach is that we can identify duplicate failures and log only once. For example, if a metric exporter failing in every iteration log file will be flooded with the same messages. With this approach we can log only for first time and log again once the issue is resolved.

Reply:
Agree. Will change accordingly.